PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1

```diff
@@ -839,25 +839,47 @@ function Get-SddcDiagnosticInfo
     try { 
 		if ($ClusterName -eq ".")
 		{
-			$Cluster = Get-Cluster -Name $ClusterNodes[0].Name
-			$ClusterName = $Cluster.Name
+			foreach ($cn in $ClusterNodes)
+			{
+				$Cluster = Get-Cluster -Name $cn[0].Name -ErrorAction SilentlyContinue
+				
+				# if we cannot connect to cluster service will still have an access node this way
+				$AccessNode = $cn[0].Name
+				
+				if ($Cluster -eq $null)
+				{
+					continue;
+				}				
+				$ClusterName = $Cluster.Name
+				break;
+			}
 		}
 		else
 		{
 			$Cluster = Get-Cluster -Name $ClusterName
+			$AccessNode = $ClusterNodes[0].Name
 		}
 	}
     catch { Show-Error("Cluster could not be contacted. `nError="+$_.Exception.Message) }
-    if ($null -eq $Cluster) { Show-Error("Server is not in a cluster") }
-    $Cluster | Export-Clixml ($Path + "GetCluster.XML")
-
-    $ClusterName = $Cluster.Name + "." + $Cluster.Domain
-    $S2DEnabled = $Cluster.S2DEnabled
-
-    # Select an access node, which will be used to query the cluster
-
-    $AccessNode = ($ClusterNodes)[0].Name + "." + $Cluster.Domain
-    Write-Host "Cluster name               : $ClusterName"
+	
+	if ($Cluster -ne $null)
+	{
+		$Cluster | Export-Clixml ($Path + "GetCluster.XML")
```

Comment:
Note GetCluster.XML is only dropped if we find a cluster.

Reply:
Yup, doesn't make sense to drop a file we couldn't possible create.